pkg/envoy/xds_server.go

```diff
@@ -829,10 +829,14 @@ func getListenerFilter(isIngress bool, useOriginalSourceAddr bool, l7lb bool) *e
 		ingressIPv4 := node.GetIngressIPv4()
 		if ingressIPv4 != nil {
 			conf.Ipv4SourceAddress = ingressIPv4.String()
+			// Enforce ingress policy for Ingress
+			conf.EnforcePolicyOnL7Lb = true
```

Comment:
Does this have any upgrade impacts?

Reply:
This should not have upgrade impacts, as long as Envoy is updated at the same time, which still is the case also with the daemonset envoy mode.